app/models/response.rb

```diff
@@ -271,7 +271,7 @@ def populate_from_j2me(data)
   # Groups answers by questioning.
   # Makes sure there are associated answer objects for each questioning in the form.
   def answer_sets
-    @answer_sets ||= visible_questionings.map{ |qing| answer_set_for_questioning(qing) }
+    @answer_sets ||= Hash[visible_questionings.map{ |qing| [qing, answer_set_for_questioning(qing)] }]
```

Comment:
I'm pretty sure you need to do it like `Hash[*foo.map{ |x| [x, bar(x)] }.flatten]` or you won't get what you want.

But even better in this situation might be to just map to the answer sets and then `index_by(&:qing)`?